assembl/static2/js/app/components/administration/voteSession/customizeGaugeForm.jsx

```diff
@@ -73,17 +75,30 @@ export class DumbCustomizeGaugeForm extends React.Component<DefaultProps, Props,
   }
 
   handleSubmit = () => {
-    const { close, customizeModule } = this.props;
+    const { cancelAllCustomizations, close, gaugeModuleId, updateModule } = this.props;
     const { gaugeParams } = this.state;
     this.setState({ saving: true });
-    customizeModule(gaugeParams);
+    if (this.state.applyToAllProposals) {
+      const { voteSpecTemplateId } = this.props;
+      const params = {
+        ...gaugeParams,
+        isCustom: false,
+        // set random ids to create new choices
+        choices: gaugeParams.choices && gaugeParams.choices.map(c => c.set('id', createRandomId()))
+      };
+      // ensure that the template is updated before to cancel the customizations
+      Promise.resolve(updateModule(voteSpecTemplateId, params)).then(() => cancelAllCustomizations(voteSpecTemplateId));
+    } else {
+      updateModule(gaugeModuleId, { ...gaugeParams, isCustom: true });
```

Comment:
I'm not sure this is a good architecture to have the same function called alternatively with voteSpecTemplateId and gaugeModuleId but why not